src/main/scala/dpla/ingestion3/mappers/providers/DcMapping.scala

```diff
@@ -0,0 +1,138 @@
+    extractStrings(data \ "metadata" \\ "date")
+      .map(stringOnlyTimeSpan)
+
+  override def description(data: Document[NodeSeq]): Seq[String] =
+    extractStrings(data \ "metadata" \\ "description") ++
+      extractStrings(data \ "metadata" \\ "source")
+
+  override def format(data: Document[NodeSeq]): Seq[String] =
+    (extractStrings(data \ "metadata" \\ "format") ++
+      extractStrings(data \ "metadata" \\ "type"))
+      .map(_.applyBlockFilter(
+          DigitalSurrogateBlockList.termList ++
+          FormatTypeValuesBlockList.termList)
+        ).filter(_.nonEmpty)
+
+  override def identifier(data: Document[NodeSeq]): Seq[String] =
+    extractStrings(data \ "metadata" \\ "identifier")
+
+  override def language(data: Document[NodeSeq]): Seq[SkosConcept] =
+    extractStrings(data \ "metadata" \\ "language")
+      .map(nameOnlyConcept)
+
+  override def place(data: Document[NodeSeq]): Seq[DplaPlace] =
+    extractStrings(data \ "metadata" \\ "place")
+      .map(nameOnlyPlace)
+
+  override def publisher(data: Document[NodeSeq]): Seq[EdmAgent] =
+    extractStrings(data \ "metadata" \\ "publisher")
+      .map(nameOnlyAgent)
+
+  override def rights(data: Document[NodeSeq]): Seq[String] =
+    (data \ "metadata" \\ "rights").flatMap(r => {
+      r.prefix match {
+        case "dc" => Option(r.text)
+        case _ => None
+      }
+    })
+
+  override def subject(data: Document[NodeSeq]): Seq[SkosConcept] =
+    extractStrings(data \ "metadata" \\ "subject")
+      .flatMap(_.splitAtDelimiter(";"))
+      .map(nameOnlyConcept)
+
+  override def title(data: Document[NodeSeq]): Seq[String] =
+    extractStrings(data \ "metadata" \\ "title")
+
+  override def `type`(data: Document[NodeSeq]): Seq[String] =
+    (extractStrings(data \ "metadata" \\ "type") ++
+      extractStrings(data \ "metadata" \\ "format"))
+    .flatMap(_.splitAtDelimiter(";"))
+
+
+  // OreAggregation
+  override def dplaUri(data: Document[NodeSeq]): URI = mintDplaItemUri(data)
+
+  override def dataProvider(data: Document[NodeSeq]): EdmAgent = {
+    extractStrings(data \ "metadata" \\ "dataProvider")
+      .map(nameOnlyAgent)
+      .headOption // take the first value
+      .getOrElse( // return the first value or throw an exception
+      throw new Exception(s"Missing required property metadata/dataProvider is empty for ${getProviderId(data)}")
+    )
+  }
+
+  override def edmRights(data: Document[NodeSeq]): ZeroToOne[URI] = {
+    (data \ "metadata" \\ "rights").flatMap(r => r.prefix match {
+      case "edm" => Option(Utils.createUri(r.text))
+      case _ => None
+    }).headOption
+  }
+
+  override def isShownAt(data: Document[NodeSeq]): EdmWebResource =
+    uriOnlyWebResource(
+      Utils.createUri(extractStrings(data \ "metadata" \\ "isShownAt")
+        .headOption
+        .getOrElse(
+          throw new RuntimeException(s"No isShownAt property in record ${getProviderId(data)}")
```

Comment:
Might make sense to have a single, purpose-built exception we always throw for mapping errors.

Reply:
Agreee